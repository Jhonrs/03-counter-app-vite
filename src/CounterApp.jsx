import { useState } from "react";
import PropsTypes from "prop-types";

export const CounterApp = ({ value }) => {
    const [counter, setCounter] = useState(value);
    const handleAdd = () => {
        setCounter(counter + 1);
        //setCounter( (c) => c + 1 );
    };
    const handleRemove = () => setCounter(counter - 1);
    const handleReset = () => setCounter(value);

    return (
        <>
            <h1>CounterApp</h1>

            <h2>{counter}</h2>
            <button data-testid="btn-increase" onClick={handleAdd}>+1</button>
            <button data-testid="btn-decrease" onClick={handleRemove}>-1</button>
            <button data-testid="btn-reset" onClick={handleReset}>Reset</button>
        </>
    );
};

CounterApp.propTypes = {
    value: PropsTypes.number.isRequired,
};
