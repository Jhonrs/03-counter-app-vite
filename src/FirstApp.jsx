import PropsTypes from "prop-types";

export const FirstApp = ({ title, subtitle }) => {
  return (
    <>
      <h2>{title}</h2>
      <h6> {subtitle}</h6>
    </>
  );
}

FirstApp.propTypes = {
  title: PropsTypes.string.isRequired,
  subtitle: PropsTypes.string.isRequired,
}

FirstApp.defaultProps = {
  subtitle: 'Template del subtitulo',
  title: 'Template del titulo',
}