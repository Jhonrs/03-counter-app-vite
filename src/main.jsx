import React from 'react';
import ReactDOM from 'react-dom/client';
import {HelloWorldApp} from './HelloWorldApp';
import {CounterApp} from './CounterApp';
import './styles.css'
import { FirstApp } from './FirstApp';

ReactDOM.createRoot(document.getElementById('root')).render(
    <React.StrictMode>
       {/*  <CounterApp value={6}/> */}
       <FirstApp title='Hola, soy Goku'/>
    </React.StrictMode>
)