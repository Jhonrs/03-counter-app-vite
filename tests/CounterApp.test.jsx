import { fireEvent, render, screen  } from "@testing-library/react";
import { CounterApp } from "../src/CounterApp";


describe('pruebas en <CounterApp />', ()=>{
    const value = 10;
    test('debe de hacer match con el snapshot', () => { 
        const { container } = render( <CounterApp value={value} />);
        expect( container ).toMatchSnapshot(); 
    });

    test('debe de mostrar el valor inicial 100', () => { 
        render( <CounterApp value={100} />);
        expect( screen.getByText(100) ).toBeTruthy();
         /* screen.debug(); */ /* sirve para ver el objeto */ 
    });

    test('debe incrementar con el boton +1', () => { 

        render( <CounterApp value={value} />);
        fireEvent.click( screen.getByTestId('btn-increase') );
        expect( screen.getByText('11') ).toBeTruthy();

    })

    test('debe decrementar con el boton -1', () => { 

        render( <CounterApp value={value} />);
        fireEvent.click(screen.getByTestId('btn-decrease') );
        screen.debug();
        expect( screen.getByText('9') ).toBeTruthy();

    })


    test('debe funcionar el botÃ³n reset', () => { 

        render( <CounterApp value={value} />);
        fireEvent.click(screen.getByTestId('btn-decrease') );
        fireEvent.click(screen.getByTestId('btn-decrease') );
        fireEvent.click(screen.getByTestId('btn-decrease') );
        fireEvent.click(screen.getByTestId('btn-reset') );
        expect( screen.getByText(value) ).toBeTruthy();

    })


});