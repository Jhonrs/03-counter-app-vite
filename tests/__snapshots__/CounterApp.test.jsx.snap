// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en <CounterApp /> debe de hacer match con el snapshot 1`] = `
<div>
  <h1>
    CounterApp
  </h1>
  <h2>
    10
  </h2>
  <button
    data-testid="btn-increase"
  >
    +1
  </button>
  <button
    data-testid="btn-decrease"
  >
    -1
  </button>
  <button
    data-testid="btn-reset"
  >
    Reset
  </button>
</div>
`;
